### PR TITLE
Testing setup

### PR DESCRIPTION
**Pull request by danieldunderfelt**
Set up Jest for testing and add a few tests. Also upgrades dependencies.

Testing brings refactors, which will also come from other PR's that integrate the new HFP api, so I think it's best to merge this now. More tests can be added later.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "plugins": ["transform-decorators-legacy", "transform-class-properties"]
```
Comment (danieldunderfelt) [inline .babelrc]: Needed for Jest.

File: package.json [hunk 2]
```diff
@@ -3,70 +3,81 @@
...
-    "serve": "^9.4.0"
+    "serve": "^10.0.2"
   },
   "scripts": {
     "docker:build": "docker build -t hsldevcom/transitlog-ui .",
     "docker:run": "docker run -d -p 3000:3000 hsldevcom/transitlog-ui",
     "start": "react-app-rewired start",
     "build": "react-app-rewired build",
     "test": "react-app-rewired test --env=jsdom",
+    "test:ci": "CI=true npm run test",
```
Comment (danieldunderfelt) [inline package.json]: The CI var needs to be set, otherwise Jest runs in watch mode.

**Comment by varpushaukka** [comment]
okay how am I supposed to run tests? npm test? and why every test fails?
 or is this pr just setup and they are failing with purpose?

` 
FAIL  src/helpers/getPrecisePositionForTime.test.js
  ● Test suite failed to run

    TypeError: environment.teardown is not a function

      at node_modules/jest-runner/build/run_test.js:230:25

 FAIL  src/helpers/getCoarsePositionForTime.test.js
  ● Test suite failed to run

    TypeError: environment.teardown is not a function

      at node_modules/jest-runner/build/run_test.js:230:25

 FAIL  src/components/map/HfpLayer.test.js
  ● Test suite failed to run

    TypeError: environment.teardown is not a function

      at node_modules/jest-runner/build/run_test.js:230:25

 FAIL  src/helpers/getDelayType.test.js
  ● Test suite failed to run

    TypeError: environment.teardown is not a function

      at node_modules/jest-runner/build/run_test.js:230:25

 FAIL  src/helpers/getJourneyId.test.js
  ● Test suite failed to run

    TypeError: environment.teardown is not a function

      at node_modules/jest-runner/build/run_test.js:230:25

Test Suites: 5 failed, 5 total
Tests:       0 total
Snapshots:   0 total
Time:        0.745s
Ran all test suites.
`

**Comment by varpushaukka** [comment]
also jest-cli wasn't included to package.json, but it should be?

**Comment by danieldunderfelt** [comment]
That's a bit weird, I'll have a look. I had that error too in the beginning. The tests should pass! And jest-cli is not needed.

**Comment by danieldunderfelt** [comment]
I fixed some tests that were broken after I merged master into this branch. I recommend that @varpushaukka delete your node_modules folder and try installing again as those errors seem to be caused by dependency mismatches.

**Pull request merged**